[action] update_info_plist: Output error message if xcodeproj can not be found

obayer opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Fixes: https://github.com/fastlane/fastlane/issues/13927

### Description
<!-- Describe your changes in detail -->
<!-- Please describe in detail how you tested your changes. -->
* Wrote a test to confirm defective behaviour
* Checked against project setup described in ticket and see if the `conversion of nil` error has been replaced by a more descriptive error message

obayer committed:
[action] Fix no implicit conversion of nil into String in update_info_plist if no explicit xcodeproj is specified (#13927)

obayer commented:
I signed it!

obayer committed:
[action] Update output error message

obayer commented on fastlane/lib/fastlane/actions/update_info_plist.rb:
```diff
@@ -14,6 +14,7 @@ def self.run(params)
 
           # Assign folder from parameter or search for xcodeproj file
           folder = params[:xcodeproj] || Dir["*.xcodeproj"].first
+          UI.user_error!("You must specify an xcodeproj path") if folder.nil?
```
Feedback addressed @janpio

janpio commented on fastlane/lib/fastlane/actions/update_info_plist.rb:
```diff
@@ -14,6 +14,7 @@ def self.run(params)
 
           # Assign folder from parameter or search for xcodeproj file
           folder = params[:xcodeproj] || Dir["*.xcodeproj"].first
+          UI.user_error!("Could not figure out your xcodeproj path. Please specify it using the xcodeproj parameter") if folder.nil?
```
```suggestion
           UI.user_error!("Could not figure out your xcodeproj path. Please specify it using the `xcodeproj` parameter") if folder.nil?
```

janpio reviewed (approved):
LGTM (with the last suggestion added)

janpio committed:
Update fastlane/lib/fastlane/actions/update_info_plist.rb

Updates output error messages as suggested by reviewer to highlight affected parameter

Co-Authored-By: obayer <oliver@bayerwire.com>

obayer committed:
[action] Updates rspec after suggested change of error message

obayer commented:
@janpio is there anything I can or should do to get this PR merged?
